fastly/custom_tls_activation.go

```diff
@@ -0,0 +1,196 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// TLSActivation represents a /tls/activations response.
+type TLSActivation struct {
+	ID            string                `jsonapi:"primary,tls_activation"`
+	Configuration *TLSConfiguration     `jsonapi:"relation,tls_configuration"` // TLSConfiguration type shared with BulkCertificate
+	Domain        *TLSDomain            `jsonapi:"relation,tls_domain"`        // TLSDomain type shared with BulkCertificate
+	Certificate   *CustomTLSCertificate `jsonapi:"relation,tls_certificate"`
+	CreatedAt     *time.Time            `jsonapi:"attr,created_at,iso8601"`
+}
+
+// ListTLSActivationsInput is used as input to the ListTLSActivations function.
+type ListTLSActivationsInput struct {
+	FilterTLSCertificateID   *string // Limit the returned activations to a specific certificate.
```

Comment:
I was following the pattern I observed of having optional fields be pointers. I didn't catch that it was only on write operations. I can change this if that's what you decide.

Reply:
@joel-g to give you some context, take a read of https://willnorris.com/2014/05/go-rest-apis-and-pointers/ -- from my understanding the issue of not using pointer references are only relevant to write/update operations, where as create/read operations are less of a concern for us (I could be mistaken of course, but that was my takeaway).

Reply:
FYI - I'm PTO tomorrow. So I'll check back in on this PR Monday. Have a great weekend and thank you for your contributions so far @joel-g

Reply:
I read the blog post. If I make "FilterTLSCertificateID" a string rather than a *string will it not filter on an empty string and return nothing when the value is not set?

I see in the Bulk TLS API it is using pointers:

https://github.com/fastly/go-fastly/blob/57854aff2f13a1958bb496db0edd9a491c465a79/fastly/platform_tls.go#L39

Reply:
Hi @joel-g apologies for the long delay in responding. 

I've had a few other projects I needed to make progress on and we also had a sudden push to get a long standing v2.0.0 branch for Go-Fastly released:
https://github.com/fastly/go-fastly/releases/tag/v2.0.0

Please see the [new migration notes](https://github.com/fastly/go-fastly#migrating-from-v1-to-v2) in the README which details the main changes (there shouldn't be anything surprising or problematic in there, just working towards a more consistent code base). 

So if you could please rebase in the `master` branch when you get a chance please.

With regards to your question...

> If I make "FilterTLSCertificateID" a string rather than a *string will it not filter on an empty string and return nothing when the value is not set?

Correct.

But as this is a _list_ instruction (in other words a READ only operation) then there is no break in expected behaviour (i.e. a consumer of this code will have to choose not to provide that information). Now compare that to omitting a non-pointer field when carrying out a WRITE operation... in that case a missing value could result in a fundamentally different behaviour.

So we're very much in favour of pointer references for optional fields when doing WRITE operations (i.e. update structs) but less concerned when it comes to READ/CREATE operations (i.e. create structs).

Hope that makes sense.

Reply:
Ok thanks @Integralist, I have pulled in the changes from master and updated the three List operations introduced in this PR to no longer use pointers.

Reply:
@Integralist @phamann any updates on this?

Reply:
Hi @joel-g thanks for your patience. I've been out on paternity leave but I'm back now and am focused on reviewing your PR and helping it over the finish line.